crates/matrix-sdk/src/room/joined.rs

```diff
@@ -633,30 +648,126 @@ impl Joined {
     ///         &mime::IMAGE_JPEG,
     ///         &mut image,
     ///         None,
+    ///         NONE_THUMBNAIL,
+    ///         None,
     ///     ).await?;
     /// }
     /// # Result::<_, matrix_sdk::Error>::Ok(()) });
     /// ```
-    pub async fn send_attachment<R: Read>(
+    pub async fn send_attachment<R: Read, T: Read>(
```

Comment:
I agree. One of the reasons I created a separate method was also thath the otherone requires the data to be `Read + Seek` , but I was also thinking of adding a `use_cache` parameter to store the data locally instead of downloading it after sending it so it will need the same traits.